### ISSUE TITLE
no-shadow new option "ignoredGlobals"

### ISSUE THREAD

**Issue by Zamralik** [eslint/eslint#13856]
**What rule do you want to change?**

no-shadow (when using "builtinGlobals": true)

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

new option "ignoredGlobals", an array of global identifiers that can be shadowed even when using "builtinGlobals": true.
Some globals are common variable identifiers and also might be confusing if used as globals.
Accessing the value by "window.name" carry more meaning than simply "name" as it's more a property of the Window class rather than a global variable.
"event" is often used in event listeners.

**Please provide some example code that this change will affect:**

```js
names.forEach(
    (name) =>
    {
        console.log(name);
    }
);

element.addEventListener(
    "foo",
    (event) =>
    {
        ...
    }
);
```

**What does the rule currently do for this code?**

"name" and "event" are disallowed because of the property name of Window.

**What will the rule do after it's changed?**

"name" and "event" could be allowed to shadow.

**Are you willing to submit a pull request to implement this change?**

I do not know how to do it myself, so no.

**Comment by anikethsaha** [comment]
Thanks for the issue. 
I think [`allow`](https://eslint.org/docs/rules/no-shadow#allow) might fix your issue. 

Let me know if I am missing anything.

**Comment by Zamralik** [comment]
> 
> 
> Thanks for the issue.
> I think [`allow`](https://eslint.org/docs/rules/no-shadow#allow) might fix your issue.
> 
> Let me know if I am missing anything.

Thanks!
